azurelinuxagent/ga/monitor.py

```diff
@@ -399,54 +409,39 @@ def send_telemetry_heartbeat(self):
-                for cgroup_name, metrics in metric_reported.items():
-                    thresholds = metric_threshold[cgroup_name]
-
-                    for metric_group, metric_name, value in metrics:
-                        if value > 0:
-                            report_metric(metric_group, metric_name, cgroup_name, value)
-
-                        if metric_group == "Memory":
-                            # Memory is collected in bytes, and limit is set in megabytes.
-                            if value >= CGroupConfigurator._format_memory_value('megabytes', thresholds.memory_limit):
-                                msg = "CGroup {0}: Crossed the Memory Threshold. " \
-                                      "Current Value: {1} bytes, Threshold: {2} megabytes." \
-                                       .format(cgroup_name, value, thresholds.memory_limit)
-
-                                logger.warn(msg)
-                                add_event(name=AGENT_NAME,
-                                          version=CURRENT_VERSION,
-                                          op=WALAEventOperation.CGroupsLimitsCrossed,
-                                          is_success=True,
-                                          message=msg,
-                                          log_event=True)
-
-                        if metric_group == "Process":
-                            if value >= thresholds.cpu_limit:
-                                msg = "CGroup {0}: Crossed the Processor Threshold. " \
-                                      "Current Value: {1}, Threshold: {2}." \
-                                       .format(cgroup_name, value, thresholds.cpu_limit)
-
-                                logger.warn(msg)
-                                add_event(name=AGENT_NAME,
-                                          version=CURRENT_VERSION,
-                                          op=WALAEventOperation.CGroupsLimitsCrossed,
-                                          is_success=True,
-                                          message=msg,
-                                          log_event=True)
-
-            except Exception as e:
-                logger.warn("Monitor: failed to collect cgroups performance metrics: {0}", ustr(e))
-                logger.verbose(traceback.format_exc())
-
-            self.last_cgroup_telemetry = datetime.datetime.utcnow()
+    def send_extension_metrics_telemetry(self):
+        to_send = False
+        time_now = datetime.datetime.utcnow()
+        performance_metrics = {}
+
+        if not self.last_cgroup_polling_telemetry:
+            self.last_cgroup_polling_telemetry = time_now
+
+        if not self.last_cgroup_report_telemetry:
+            self.last_cgroup_report_telemetry = time_now
+
+        if time_now >= (self.last_cgroup_polling_telemetry +
+                        MonitorHandler.CGROUP_TELEMETRY_POLLING_PERIOD):
+            CGroupsTelemetry.poll_all_tracked()
```

Comment:
The names for poll_all_tracked and collect_all_tracked are too similar... it is hard to see at a glance what they are doing. 

I think CGROUP_TELEMETRY_POLLING_PERIOD and CGROUP_TELEMETRY_REPORTING_PERIOD express very clearly their intent. Can we rename to include "reporting" somewhere?